docs/readme-zephyr.md

```diff
@@ -209,3 +209,57 @@ swap-using-scratch is used whereby an application has loaded a firmware update
 and marked it as test/confirmed but MCUboot will not swap the images and
 erasing the secondary slot from the zephyr application returns an error
 because the slot is marked for upgrade.
+
+## Serial recovery
+
+### Interface selection
+
+A serial recovery protocol is available over either a hardware serial port or an USB CDC ACM virtual serial port.
+The SMP server implementation can be enabled by ``CONFIG_MCUBOOT_SERIAL=y``.
+To set a type of an interface, use ``BOOT_SERIAL_DEVICE`` Kconfig choice, and select either ``CONFIG_BOOT_SERIAL_UART`` or ``CONFIG_BOOT_SERIAL_CDC_ACM`` value.
+Which interface belongs to the protocol shall be set by the devicetree-chosen node:
+- `zephyr,console` - If a hardware serial port is used.
+- `zephyr,cdc-acm-uart` - If a virtual serial port is used.
+
+### Entering the serial recovery mode
+
+To enter the serial recovery mode, the device has to initiate rebooting, and a triggering event has to occur (for example, pressing a button).
+
+By default, the serial recovery GPIO pin active state enters the serial recovery mode.
+Use the ``mcuboot_button0`` devicetree button alias to assign the GPIO pin to the MCUboot.
```

Comment:
Note that all of this will become outdated once https://github.com/mcu-tools/mcuboot/pull/1629 is merged